scripts/fig1-sample_disease_barchart.R

```diff
@@ -0,0 +1,167 @@
+
+# all metadata files 
+sample_info_dir <- file.path(root_dir, "sample-info")
+project_whitelist_file <- file.path(sample_info_dir, "project-whitelist.txt")
+diagnosis_groupings_file <- file.path(sample_info_dir, "diagnosis-groupings.tsv")
+disease_timing_file <- file.path(sample_info_dir, "disease-timing.tsv")
+
+# output files 
+plots_dir <- file.path(root_dir, "figures", "pngs")
+output_plot_file <- file.path(plots_dir, "Fig1-sample-summary.png")
+
+# Prep sample metadata ------------------------------------------------------
+
+# read in project whitelist and grouping metadata 
+project_whitelist <- readLines(project_whitelist_file)
+diagnosis_groupings_df <- readr::read_tsv(diagnosis_groupings_file) |>
+  dplyr::select(submitted_diagnosis, diagnosis_group)
+disease_timing_df <- readr::read_tsv(disease_timing_file)
+
+# Copy sample metadata 
+local_sample_metadata <- file.path(local_s3_dir, "scpca-sample-metadata.tsv")
+sync_call <- paste('op run -- aws s3 cp', sample_metadata_s3, local_sample_metadata, '--recursive', sep = " ")
+system(sync_call)
+
+# read in sample metadata and filter to only projects in whitelist
+sample_metadata_df <- readr::read_tsv(local_sample_metadata) |> 
+  dplyr::filter(scpca_project_id %in% project_whitelist)
+
+# Join sample metadata with diagnosis grouping and disease timing 
+diagnosis_plot_df <- sample_metadata_df |> 
+  dplyr::select(scpca_sample_id, diagnosis, disease_timing) |> 
+  dplyr::left_join(diagnosis_groupings_df, by = c("diagnosis" = "submitted_diagnosis")) |> 
+  dplyr::left_join(disease_timing_df, by = c("disease_timing" = "submitted_disease_timing")) |> 
+  # right now let's remove this group
+  dplyr::filter(diagnosis_group != "Non-cancerous") |> 
+  # make it a factor so we can use ggpattern 
+  dplyr::mutate(standardized_disease_timing = as.factor(standardized_disease_timing))
+
+
+# get list of all groups 
+diagnosis_groups <- diagnosis_plot_df |> 
+  dplyr::pull(diagnosis_group) |>
+  unique() |>
+  # set desired order 
+  forcats::fct_relevel("Brain and CNS",
+                       "Leukemia", 
+                       "Sarcoma",
+                       "Other solid tumors")
+# colors 
+# TODO: move these into a separate tsv file to be read in 
+#names(diagnosis_groups) <- c("#201158", "#005E5E", "#578B21", "#E89E6B")
+names(diagnosis_groups) <- c("#DF536B", "#61D04F", "#2297E6", "#28E2E5")
+
+# Plot -------------------------------------------------------------------------
+
+# Create list of plots that will be combined into one figure 
+all_panels <- diagnosis_groups |> 
+  purrr::imap(\(group, group_color){
+    barplot_panel(diagnosis_plot_df = diagnosis_plot_df,
+                  diagnosis_group = group, 
+                  group_color = group_color)
+  })
+
+# grab one legend from list of plots to use as the common legend
+legend_to_use <- ggpubr::get_legend(all_panels[[1]])
+
+# combine into one plot using a common legend 
+combined_plot <- ggpubr::ggarrange(plotlist = all_panels,
```

Comment:
Is there a reason you are using `ggarrange` rather than faceting on group? I feel like that would probably solve the legend color issue.

I think if you sort the diagnoses first by group and then use `scale=free` in the facet, you should be able to get around the label ordering issue you might have been worried about)

Reply:
Hmm I did use `scale_x=free` and `scale_y=free`,  but that only worked with creating a single row or single column. I wasn't able to get `facet_wrap` to work as expected, but I can try it again and make sure I use `scale=free`. The issue with faceting was also getting the patterns to work correctly, but I'll play around with it again.

Reply:
A semi-minimal example that I think gets all the things you want?
```
library(ggplot2)
df <- data.frame(
  group = c("A", "A", "A", "B", "B", "B", "B", "B"),
  diagnosis = c("A1", "A1", "A2", "B1", "B1", "B1", "B2", "B2"),
  timepoint = c("x", "y", "x", "x", "x", "z", "z", "y")
)

ggplot(df, aes(y = diagnosis,  fill = group)) +
  ggpattern::geom_bar_pattern(aes(pattern = timepoint), 
                              color = "black",
                              pattern_color = "black",
                              pattern_fill = "black") +
  facet_wrap(facets = "group", scales = "free") +
  ggpattern::scale_pattern_manual(
    values = c("x" = "none", "y" = "stripe", "z" = "crosshatch")) + 
  guides(fill = "none", 
         pattern = guide_legend(override.aes = list(fill = "white"))) + 
  labs(pattern = "Time Point") + 
  theme(text = element_text(size = 10),
        legend.position = "top",
        legend.key.size = unit(1, 'cm')
  )
```